### ISSUE TITLE
When a node is down, Alternator TTL could continue to expire items owned by this node

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#9787]
The current implementation of the Alternator expiration (TTL) feature assigns to each node a subset of the token range (the _primary_ ranges of this node), and the node scans just its assigned ranges. This means that if a single node is down, nobody will check the items in its ranges for expiration. 

The point of this issue is to note that it doesn't have to be this way: If only a single node is down, and RF=3, the items that node owns are still readable with QUORUM - so these items can still be safely read and checked for expiration - and also deleted.

There's a fairly simple way we can implement that: When a node completes scanning its own primary ranges - or in parallel with it doing so - the node can also check whether any other node is down - and there is such a node, whether any of this node's replicated ranges have: 1. the dead node as their primary replica, and 2. the current node as the second (out of 3) replica. For such ranges, this node will scan them as well. This secondary scan stops if the remote node comes back up.

Later we can further improve on the above plan by using a global table (and perhaps LWT) for "locking" ranges that are being scanned, to avoid the same range being scanned by both primary and secondary owner - although the damage from that ever happening are slim (basically, the possible deletion of the same item twice, which doesn't matter except perhaps for Alternator Streams). If we do have such a table, we can also use it to keep track of *when* each range was scanned. This can let us guarantee that we scan each range once every X hours, instead of the current situation that expiration of an item may be delayed to 2*X if the scan is restarted in the middle, or even more if the scan is restarted multiple times.

@avikivity noted that this issue is of **low importance**: A cluster which has a node down is at risk anyway - any other node going down will cause unavailability (for QUORUM writes or reads) - so the user must act quickly and either bring this node back up, or take it out of the cluster. Since the user is expected to take one of these actions quickly, the scanning of this range will resume quickly one way or another when the range returns to being owned by some node. (note that there is a separate issue of how Alternator TTL behaves after a cluster topology change - it can further delay the time until items are expired).

I wrote a dtest, `alternator_ttl.py:AlternatorTTLTest.test_expiration_with_down_node`, to reproduce this issue. The test starts a 5-node cluster, with 1000 items with random tokens which are due to be expired immediately. The test expects to see all items expiring ASAP, but when one of the five nodes is brought down, this doesn't happen: Some of the items are not expired.
